includes/class-wc-gateway-stripe.php

```diff
@@ -1035,11 +1051,19 @@ public function verify_intent_after_checkout( $order ) {
 				$order->payment_complete();
 			}
 		} else if ( 'succeeded' === $intent->status || 'requires_capture' === $intent->status ) {
+			do_action( 'wc_stripe_before_intent_success', $order );
```

Comment:
I'm personally not a fan of using actions/filters to pass messages on the same plugin. Since `WC_Stripe_Subs_Compat` extends `WC_Gateway_Stripe`, what do you think about defining some empty methods in the parent class (`handle_intent_success_verification($order)`, `handle_intent_sca_failure($order)`) and overriding them in the subs compat class?

Reply:
Fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1084/commits/43427444cf7a7e7f31cc34d71ab7884f9c8f9faa. I was also not very confident in those filters.

Reply:
What do you think about https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1084/commits/546349a88ba6118ae6e235b5e9a1696746f97d80? (I can delete the commit if you don't like it). I found a bit confusing that the `success` callback was called *after* the default behaviour, but the `failure` callback was called *before*.

Reply:
The commit is 💯 . Then there's me looking at it and wondering why I didn't do that in the first place 🤦‍♂